code/fighter.gd

```diff
@@ -151,8 +151,9 @@ func process_hit(EnteredHitbox, HitSide):
 
 func teleport_and_project():
 	#(K)Teleport…
-	var DeltaFreezePos = HitlagTeleportDelta * FreezeInput;
-	move(DeltaFreezePos)
+	var DeltaHitPos = 20 * BulletDir # NOTE(hugo) : the fighter should advance in the direction it shoots towards
+	var DeltaFreezePos = HitlagTeleportDelta * FreezeInput
+	move(DeltaFreezePos + DeltaHitPos)
```

Comment:
_Functional issue : _
Actually my wording of the ticket was not clear : by saying the hit has the fighter "décalé" I meant it to ~~teleport~~ **project** towards the front.
So you may undo the changes here and make the following change, later in the file : 
```
	if (LastHitSide == "left"):
		HitDirection = Vector2(0, 1) + 0.5 * BulletDir
	else: # LastHitSide == "right"
		HitDirection = Vector2(0, -1) + 0.5 * BulletDir
```